api/src/services/import-services/capture/capture-header-configs.ts

```diff
@@ -0,0 +1,50 @@
+import { z } from 'zod';
+import { CSVConfigUtils } from '../../../utils/csv-utils/csv-config-utils';
+import { CSVCellValidator } from '../../../utils/csv-utils/csv-config-validation.interface';
+import { validateZodCell } from '../../../utils/csv-utils/csv-header-configs';
+import { ICritterDetailed } from '../../critterbase-service';
+import { findCapturesFromDateTime } from '../utils/datetime';
+import { CaptureCSVStaticHeader } from './import-captures-service';
+
+/**
+ * Get the capture date cell validator.
+ *
+ * Rules:
+ *  1. The cell combined with the 'CAPTURE_TIME' must be a valid timestamp
+ *  2. The timestamp must NOT map to an existing capture for the critter
+ *
+ * @param {Map<string, ICritterDetailed>} surveyAliasMap The survey alias map
+ * @param {CSVConfigUtils<MarkingCSVStaticHeader>} utils The CSV config utils
+ * @returns {*} {CSVCellValidator} The validate cell callback
+ */
+export const getCaptureDateCellValidator = (
+  surveyAliasMap: Map<string, ICritterDetailed>,
+  utils: CSVConfigUtils<CaptureCSVStaticHeader>
+): CSVCellValidator => {
+  return (params) => {
+    const cellErrors = validateZodCell(params, z.string().date());
+
+    if (cellErrors.length) {
+      return cellErrors;
+    }
+
+    // Row meta data
+    const captureDate = String(params.cell);
+    const captureTime = utils.getCellValue('CAPTURE_TIME', params.row) as string; // casting to allow undefined
```

Comment:
Should this be `as string | undefined;`?

Reply:
Fixed